dimod/reference/composites/clipcomposite.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2019 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# =============================================================================
+"""
+A composite that clips problem variables below and above threshold. if lower and upper bounds is not given
+it does nothing.
+
+"""
+try:
+    import collections.abc as abc
+except ImportError:
+    import collections as abc
+
+
+from dimod.binary_quadratic_model import BinaryQuadraticModel
+from dimod.core.composite import ComposedSampler
+from dimod.sampleset import SampleSet
+
+__all__ = 'ClipComposite',
```

Comment:
@randomir has given me a hard time for this before :smile: preferred pattern is `__all__ = ['ClipComposite']`